flytepropeller/pkg/compiler/transformers/k8s/workflow.go

```diff
@@ -244,7 +249,19 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 		errs.Collect(errors.NewWorkflowBuildError(err))
 	}
 
-	obj.ObjectMeta.Name = GetExecutionName(name, time.Now().UnixNano())
+	hashedIdentifier := hashIdentifier(core.Identifier{
+		Project: project,
+		Domain:  domain,
+		Name:    name,
+	})
+	rand.Seed(int64(hashedIdentifier))
+
+	if workflowCRNameHashLength := config.GetConfig().WorkflowCRNameHashLength; workflowCRNameHashLength > 0 {
+		obj.ObjectMeta.Name = rand.String(workflowCRNameHashLength)
+	} else {
+		obj.ObjectMeta.Name = name
+	}
```

Comment:
By default `WorkflowCRNameHashLength` is 0 so the hash CR naming scheme will only be used when the deployment is specially configured to do so.